### PR TITLE
Feature/trainer

### PR DESCRIPTION
**Pull request by alstjrdlzz**
loss.py
---
nn.BCELoss에 잘못된 arguments를 입력하던 오류를 수정했습니다.

data/dataset.py
---
기존 TrainDataset은 getitem method에서 넘파이 변환, 인덱싱, tenosr 변환을 수행했습니다. 
getitem은 batch를 만들때마다 호출되는 함수이므로 위 작업을 getitem 단에서 수행하는 것은 bottleneck을 유발했습니다.
따라서, 위 작업들을 init 단에서 수행하도록 수정했습니다. 위 작업들은 이제 처음 train_dataset을 선언할 때 한번만 수행되게 됩니다.
추가로 nn.BCELoss는 float type의 tensor를 입력으로 받는 것을 확인하여, 
tensor로 변환할 때 dtype을 추가하여 X는 long tensor, y는 float tensor로 변환하도록 수정했습니다.

train.py
---
단순 오타가 몇군데 확인되어 수정한 후 테스트했습니다.

model/GMF.py
---
기존 forward 함수에 오류가 있어 수정했습니다.
수정 후 forward 함수는 입력데이터를 torch.split를 이용해 user와 item으로 구분하고 각각 임베딩한 후 torch.mul을 이용해 elementwise 곱을 수행합니다. 그후 1개 Linear 레이어와 sigmoid 레이어를 거친 후 확률값을 반환합니다. 이때, 차원은 (batch_size, 1) 입니다.

### CODE REVIEW

**Comment by alstjrdlzz** [comment]
내일 colab GPU 용량이 확보되면 
tqdm으로 batch 하나를 만드는데 걸리는 시간을 측정하는 실험을 진행하시고 
이전과 비교하여 bottleneck이 어느정도 해소됐는지 확인해주세요
